tests/test_2025_07_25_hoodi_dg_upgrade.py

```diff
@@ -0,0 +1,320 @@
+    # =======================================================================
+    # START VOTE
+    vote_id = (
+        vote_ids_from_env[0] if vote_ids_from_env else start_vote({"from": ldo_holder, "force": True}, silent=True)[0]
+    )
+
+    last_dual_governance_proposal_id = timelock.getProposalsCount()
+    vote_tx: TransactionReceipt = helpers.execute_vote(vote_id=vote_id, accounts=accounts, dao_voting=contracts.voting)
+
+    vote_events = group_voting_events_from_receipt(vote_tx)
+
+    expected_dg_proposal_id = last_dual_governance_proposal_id + 1
+
+    assert expected_dg_proposal_id == timelock.getProposalsCount()
+
+    # =======================================================================
+    # ==================== DG Proposal Submit => Execute ====================
+    # =======================================================================
+
+    # EXECUTE DUAL GOVERNANCE PROPOSAL
+    chain.sleep(timelock.getAfterSubmitDelay() + 1)
+
+    dual_governance.scheduleProposal(expected_dg_proposal_id, {"from": stranger})
+
+    chain.sleep(timelock.getAfterScheduleDelay() + 1)
+
+    dg_tx: TransactionReceipt = timelock.execute(expected_dg_proposal_id, {"from": stranger})
+
+    display_dg_events(dg_tx)
+    dg_events = group_dg_events_from_receipt(dg_tx, timelock=TIMELOCK, admin_executor=ADMIN_EXECUTOR)
+
+    # =======================================================================
+    # ================= After DG proposal execution tests ===================
+    # =======================================================================
+
+    assert timelock.getGovernance() == NEW_DUAL_GOVERNANCE
+    assert dual_governance.getConfigProvider() == CONFIG_PROVIDER_FOR_DISCONNECTED_DUAL_GOVERNANCE
+
+    assert new_dual_governance.getConfigProvider() == ACTIVE_CONFIG_PROVIDER
+    assert new_dual_governance.getResealCommittee() == RESEAL_COMMITTEE
+    assert new_dual_governance.getProposalsCanceller() == VOTING
+    proposers = new_dual_governance.getProposers()
+    assert len(proposers) == 1
+    assert proposers[0][0] == VOTING
+    assert proposers[0][1] == ADMIN_EXECUTOR
+    assert new_dual_governance.getTiebreakerDetails() == (
+        False,
+        NEW_TIEBREAKER_COMMITTEE,
+        TIEBREAKER_ACTIVATION_TIMEOUT,
+        [WITHDRAWAL_QUEUE, VALIDATORS_EXIT_BUS_ORACLE, TRIGGERABLE_WITHDRAWALS_GATEWAY],
+    )
+
+    # Test that old DG can't submit proposals
+    with reverts("CallerIsNotGovernance: 0x4d12b9f6acab54ff6a3a776ba3b8724d9b77845f"):
+        dual_governance.submitProposal([], "This should revert", {"from": VOTING})
+
+    # Test that old escrow won't get into Rage Quit if lock amount above previous 2nd seal
+    old_escrow = interface.DualGovernanceEscrow(contracts.dual_governance.getVetoSignallingEscrow())
+    new_escrow = interface.DualGovernanceEscrow(new_dual_governance.getVetoSignallingEscrow())
+
+    old_dual_governance_second_seal = interface.IDualGovernanceConfigProvider(
```

Comment:
Namin is confusing. Actually, dual_governance.getConfigProvider() already refers to the new config value, which is 100%

Reply:
`old_dual_governance_second_seal` variable is not defined